### ISSUE TITLE
OSMP FMU linker error

### ISSUE THREAD

**Issue by ClemensLinnhoff** [esmini/esmini#638]
Using the current release 2.40.4, I get a linker error, when running the OSMP FMU. This comes from the newly added spdlog library, as it is not linked in the FMU CMakeLists.

I could solve this as a quick-and-dirty fix by adding the following line to the link_directories in the CMakeLists of the FMU:

```
${CMAKE_CURRENT_SOURCE_DIR}/../build/externals/spdlog
```

But this links to the build folder of esmini and is therefore quite the hack.

@eknabe do you have a better solution to link the built spdlog in the OSMP FMU?

**Comment by eknabevcc** [comment]
Not fully understanding the issue. Is it compile time linking or runtime linking that fails?

I would assume the spdlog dependency would be solved by linking esmini lib, which should include spdlog. But spdlog is a new addition to esmini and there might be some baby issues we did not foresee.

I checked the CMakeLists.txt file and did some rough naïve cleaning. It compiles, but I have no clue if it runs.

For our investigation, what is the simplest possible way to test-run the FMU? Assume a total FMI/FMU noob :)
